src/main/java/Duke/Main/Parser.java

```diff
@@ -0,0 +1,58 @@
+package duke.main;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import duke.errors.InvalidCommandException;
+import duke.errors.DateFormatException;
+
+/**
+ * The type Parser.
+ */
+public class Parser {
+    /**
+     * Converts a date string into LocalDateTime object
+     *
+     * @param by the by
+     * @return LocalDateTime object
+     */
+    public static LocalDateTime strToDate(String by) throws Exception {
```

Comment:
I like this method's design and comments provided! A smart design for date parser. 👍